impeller/core/host_buffer.h

```diff
@@ -95,6 +95,22 @@ class HostBuffer final : public std::enable_shared_from_this<HostBuffer>,
                                    size_t length,
                                    size_t align);
 
+  using EmplaceProc = std::function<void(uint8_t* buffer)>;
+
+  //----------------------------------------------------------------------------
+  /// @brief      Emplace non-uniform data (like contiguous vertices) onto the
+  ///             host buffer directly. The buffer ptr is guaranteed to have at
+  ///             least enough storage for the length of data.
+  ///
+  /// @param[in]  cb            A callback that will be passed a ptr to the
+  ///                           underlying host buffer.
+  ///
+  /// @return     The buffer view.
+  ///
+  BufferView EmplaceCallback(size_t length,
```

Comment:
Nit: I'd rather just take advantage of function overloading and use the name `Emplace`.

Reply:
I suspect we will only use this sparingly for performance but its good to have the open as needed.